packages/components/src/components/popover/popover.jsx

```diff
@@ -4,223 +4,214 @@ import React from 'react';
 import TinyPopover, { ArrowContainer } from 'react-tiny-popover';
 import Icon from '../icon';
 
-class Popover extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = {
-            is_open: false,
-            is_bubble_open: false,
-            popover_ref: undefined,
-            is_mounted: false,
-        };
-
-        this.setWrapperRef = ref => {
-            this.wrapper_ref = ref;
-            this.setState({ popover_ref: ref });
-        };
-    }
-
-    componentDidMount() {
-        this.setState({ is_open: this.props.has_error, is_mounted: true });
-    }
-
-    toggleOpen = () => {
-        this.setState({ is_open: Boolean(this.props.message) });
-    };
-
-    toggleClose = () => {
-        if (this.props.is_bubble_hover_enabled) {
-            this.toggleIsOpenOnHoverPopoverBubble();
+const Popover = ({ ...props }) => {
+    const ref = React.useRef();
+
+    const [is_popover_open, setIsPopoverOpen] = React.useState(false);
+    const [is_bubble_open, setIsBubbleOpen] = React.useState(false);
+    const [popover_ref, setPopoverRef] = React.useState(undefined);
+
+    const has_external_open_state = props.is_open !== undefined;
+
+    React.useEffect(() => {
+        if (ref.current) {
+            setPopoverRef(ref.current);
+        }
+        setIsPopoverOpen(!!props.has_error);
```

Comment:
`has_error` is marked as boolean in PropTypes. I don't think we need double negation here

Reply:
It's setting the state as `undefined` when there's no `boolean` or `prop` passed. 
Double negation is there to ensure it's always passing a `bool` as i'd rather not set it as a `defaultProp`.